File: Telefonski_imenik.py
class Contact:
    def __init__(self, name, phone_number):
        self.name = name
        self.phone_number = phone_number

contacts = []

def add_contact():
    name = input("Unesite ime i prezime: ")
    phone_number = int(input("Unesite telefonski broj: "))
    new_contact = Contact(name, phone_number)
    contacts.append(new_contact)
    print("Kontakt dodan.")

def edit_contact():
    index = int(input("Unesite indeks kontakta za izmjenu: "))
    if 0 <= index < len(contacts):
        new_name = input("Unesite novo ime i prezime: ")
        new_phone_number = int(input("Unesite novi telefonski broj: "))
        contacts[index].name = new_name
        contacts[index].phone_number = new_phone_number
        print("Kontakt izmijenjen.")
    else:
        print("Indeks nije valjan.")

def delete_contact():
    index = int(input("Unesite indeks kontakta za brisanje: "))
    if 0 <= index < len(contacts):
        del contacts[index]
        print("Kontakt obrisan.")
    else:
        print("Indeks nije valjan.")

def list_contacts():
    if not contacts:
        print("Telefonski imenik je prazan.")
        return
    print("Popis kontakata:")
    for i, contact in enumerate(contacts):
        print(f"[{i}] Ime: {contact.name}, Broj: {contact.phone_number}")

def main():
    while True:
        print("Odaberite opciju:")
        print("1. Dodaj kontakt")
        print("2. Izmijeni kontakt")
        print("3. Obriši kontakt")
        print("4. Pregled svih kontakata")
        print("5. Izlaz")
        choice = int(input("Vaš odabir: "))

        if choice == 1:
            add_contact()
        elif choice == 2:
            edit_contact()
        elif choice == 3:
            delete_contact()
        elif choice == 4:
            list_contacts()
        elif choice == 5:
            print("Hvala na korištenju!")
            break
        else:
            print("Neispravan odabir. Pokušajte ponovno.")

if __name__ == "__main__":
    main()

File: Telefonski_imenik.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

struct Contact {
    string name;
    int phoneNumber;
};

vector<Contact> contacts;

void addContact() {
    Contact newContact;
    cout << "Unesite ime i prezime: ";
    cin.ignore(); 
    getline(cin, newContact.name);
    cout << "Unesite telefonski broj: ";
    cin >> newContact.phoneNumber;
    contacts.push_back(newContact);
    cout << "Kontakt dodan." << endl;
}

void editContact() {
    int index;
    cout << "Unesite redni broj kontakta za izmjenu: ";
    cin >> index;
    if (index >= 0 && index < contacts.size()) {
        cout << "Unesite novo ime i prezime: ";
        cin.ignore(); 
        getline(cin, contacts[index].name);
        cout << "Unesite novi telefonski broj: ";
        cin >> contacts[index].phoneNumber;
        cout << "Kontakt izmijenjen." << endl;
    } else {
        cout << "Redni broj nije valjan." << endl;
    }
}

void deleteContact() {
    int index;
    cout << "Unesite redni broj kontakta za brisanje: ";
    cin >> index;
    if (index >= 0 && index < contacts.size()) {
        contacts.erase(contacts.begin() + index);
        cout << "Kontakt obrisan." << endl;
    } else {
        cout << "Redni broj nije valjan." << endl;
    }
}

void listContacts() {
    if (contacts.empty()) {
        cout << "Telefonski imenik je prazan." << endl;
        return;
    }
    cout << "Popis kontakata:" << endl;
    for (size_t i = 0; i < contacts.size(); ++i) {
        cout << "[" << i << "] Ime: " << contacts[i].name << ", Broj: " << contacts[i].phoneNumber << endl;
    }
}

int main() {
    int choice;

    while (true) {
        cout << "Odaberite opciju:" << endl;
        cout << "1. Dodaj kontakt" << endl;
        cout << "2. Izmijeni kontakt" << endl;
        cout << "3. Obriši kontakt" << endl;
        cout << "4. Pregled svih kontakata" << endl;
        cout << "5. Izlaz" << endl;
        cout << "Vaš odabir: ";
        cin >> choice;

        switch (choice) {
            case 1:
                addContact();
                break;
            case 2:
                editContact();
                break;
            case 3:
                deleteContact();
                break;
            case 4:
                listContacts();
                break;
            case 5:
                cout << "Hvala na korištenju!" << endl;
                return 0;
            default:
                cout << "Neispravan odabir. Pokušajte ponovno." << endl;
        }
    }

    return 0;
}
